File: src/main/java/com/mario6/paper/repository/ProjectRepository.java
package com.mario6.paper.repository;

import com.mario6.paper.model.Project;

import java.util.List;

/**
 * ProjectRepository
 *
 * @author Mario Luo
 * @date 2019.01.19 11:23
 */
public interface ProjectRepository {

    Project findOne(String id);

    int save(Project project);

    List<Project> findAll();

    int delete(String id);
}

File: src/main/kotlin/repository/ProjectRepository.kt
package com.github.lkqm.paper.repository

import com.github.lkqm.paper.model.Project

/**
 * 项目持久层
 */
interface ProjectRepository {

    fun findOne(id: String): Project?

    fun save(project: Project): Int

    fun findAll(): List<Project>

    fun delete(id: String): Int

}